tfx_addons/firebase_publisher/README.md

```diff
@@ -21,25 +21,60 @@ This project helps users to publish trained models directly from TFX Pusher comp
-  - The URI of the pushed model from TFX Pusher component.
-  - Requirements from Firebase ML (credential JSON file path, Firebase temporary-use GCS bucket). Please find more information from [Before you begin section](https://firebase.google.com/docs/ml/manage-hosted-models#before_you_begin) in the official Firebase document.
-  - Meta information to manage published model for Firebase ML such as `display name` and `tags`.
-- Download the Firebase credential file and pushed TFLite model file.
-- Initialize Firebase Admin with the credential and Firebase temporary-use GCS bucket.
-- Search if any models with the same `display name` has already been published.
-  - if yes, update the existing Firebase ML mode, then publish it
-  - if no, create a new Firebase ML model, then publish it
-- Return `tfx.dsl.components.OutputDict(result=str)` to indicate if the job went successful, and if the job was about creating a new Firebase ML model or updating the exisitng Firebase ML model.
+- This component behaves similar to [Pusher](https://www.tensorflow.org/tfx/api_docs/python/tfx/v1/components/Pusher) component, but it pushes/hosts model to Firebase ML instead. To this end, FirebasePublisher interits Pusher, and it gets the following inputs
+
+```python
+FirebasePublisher(
+  model: types.BaseChannel = None,
+  model_blessing: Optional[types.BaseChannel] = None,
+  custom_config: Optional[Dict[str, Any]] = None,
+)
+```
+
+- Each inputs:
+  - `model` : the model from the upstream TFX component such as [Trainer](https://www.tensorflow.org/tfx/api_docs/python/tfx/v1/components/Trainer)
+  - `model_blessing` : the output of `blessing` from the Evaluator component to indicate if the given `model` is good enough to be pushed
+  - `custom_config` : additional information to initialize and configure [Firebase Admin SDK](https://firebase.google.com/docs/reference/admin/python). `FIREBASE_ML_MODEL_NAME` and `FIREBASE_ML_MODEL_TAGS` correspond to the `display_name` and `tags` respectively of the Firebase hosted model. `FIREBASE_CREDENTIALS` is a optional parameter, and it indicates GCS location where a Service Account Key (JSON) file is stored. If this parameter is not given, [Application Default Credentials](https://cloud.google.com/docs/authentication/production) will be used in GCP environment
+
+    ```python
+    FIREBASE_ML_ARGS = {
+        "FIREBASE_ML": {
+            "FIREBASE_CREDENTIALS": ...,
+            "FIREBASE_ML_MODEL_NAME": ...,
+            "FIREBASE_ML_MODEL_TAGS": ["tag1", ... ],
+            "OPTIONS": { 
+                # to be passed directly into the options argument of firebase_admin.initialize_app. 
+                # The mandatory option is storageBucket where Firebase ML temporarily stores model
+                # https://firebase.google.com/docs/reference/admin/python/firebase_admin#initialize_app
+
+                "storageBucket": ...,
```

Comment:
for this you may want to instead use the output PushedModel location instead of asking user to provide a bucket. Or is this a bucket Firebase specific?

Reply:
yes this is Firebase specific. For now, it doesn't let us host model without this.

Reply:
Can you add a reference to what does storageBucket refer to?

Reply:
It seems the document itself is not enough but the source maybe
: https://github.com/firebase/firebase-admin-python/blob/44a8fde5672828232ffa68267a71eedb270dbb16/firebase_admin/ml.py#L491

`storageBucket` is the key to use when initializing an app, and it is going to be used in the next step of `TFLiteGCSModelSource.from_tflite_model_file` or `from_saved_model`. As you see, it uploads files to the designated GCS Bucket. 

I hope this can refer to the output from the Trainer, but it seems like it has to upload. But I maybe wrong about the word `temporary`, this could be where the actual model is hosted when the model is published in Firebase.

Reply:
So it seems you can just provide it a `gcs_tflite_uri` isnt that correct? https://github.com/firebase/firebase-admin-python/blob/44a8fde5672828232ffa68267a71eedb270dbb16/firebase_admin/ml.py#L478

That said, I agree it may be good to ask user to provide one if they want to overwrite what gets used